pylint/plugins/hass_enforce_class_module.py

```diff
@@ -68,11 +68,19 @@ def visit_classdef(self, node: nodes.ClassDef) -> None:
         # we only want to check components
         if not root_name.startswith("homeassistant.components."):
             return
+        parts = root_name.split(".")
+        current_module = parts[3] if len(parts) > 3 else ""
 
         ancestors: list[ClassDef] | None = None
 
         for match in _MODULES:
-            if root_name.endswith(f".{match.expected_module}"):
+            # Allow module.py
+            # Allow module_*.py, such as light_entities.py
```

Comment:
I'd prefer if we can standardize it so that the platform module becomes a package when needed. Eg `/sensor.py` becomes `/sensor/__init__.py` with whatever submodules that are needed.

Reply:
> I'd prefer if we can standardize it so that the platform module becomes a package when needed. Eg `/sensor.py` becomes `/sensor/__init__.py` with whatever submodules that are needed.

Adjusted accordingly